aspnetcore/mvc/models/validation.md

```diff
@@ -106,6 +107,30 @@ builder.Services.AddControllers(
     options => options.SuppressImplicitRequiredAttributeForNonNullableReferenceTypes = true);
 ```
 
+<!-- remove this section in the .NET 7 version See  https://github.com/dotnet/AspNetCore.Docs/issues/25840 -->
+
+<a name="ngen"></a>
+
+## Non-nullable generic types and [Required] attribute
+
+Non-nullable generic types must include the `[Required]` attribute when the type is required. In the following code, `TestRequired` is not required:
```

Comment:
I don't know exactly what is the best wording but  I think we could change to explicitly say about the "non-nullable properties on generic types" , instead of "on generic types"